### DISCUSSION TITLE
Warnings compiling libsodium

### DISCUSSION THREAD

**Discussion by michelp** [jedisct1/libsodium#1247] [Q&A] [answered]
Hello,

I'm getting several `-Wstringop-overread` warnings in the pwhash_str api while compiling my tests for pgsodium:

https://github.com/michelp/pgsodium/actions/runs/3847667847/jobs/6554414316#step:3:6792

For example:
```
In function 'str_tests',
    inlined from 'xmain' at pwhash_argon2i.c:404:5:
pwhash_argon2i.c:293:9: warning: 'crypto_pwhash_argon2i_str_verify' reading 128 bytes from a region of size 80 [-Wstringop-overread]
  293 |     if (crypto_pwhash_argon2i_str_verify("$argon2i$m=65536,t=2,p=1c29tZXNhbHQ"
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  294 |                                          "$9sTbSlTio3Biev89thdrlKKiCaYsjjYVJxGAL3swxpQ",
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  295 |                                          "password", 0x100000000ULL) != -1) {
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

These appear to be test failures so I'm not certain it's a big deal.

-Michel

**Comment by jedisct1** [comment, marked as answer]
The warning can be safely ignored.

The function prototype indicates that the expected string length is `crypto_pwhash_STRBYTES`, but this test explicitly check that we don't crash and just return an error if the actual string is too short.

In 089f850608737f9d969157092988cb274fe7f8d4, the function prototype was changed to remove the length. That gets rid of the warning, doesn't change anything else. I'm not sure that it should be backported to `stable` since it changes a function prototype.

**Reply by michelp** [reply]
Thanks!
